### PR TITLE
Adding support for Ubuntu 22.04 image

### PR DESCRIPTION
**Pull request by IsaacCalligeros95**
Bumping from base Ubuntu 18.04 image to 22.04. Incompatible tools have been bumped to their latest available version with ping added, as it's been removed as a default tool in Ubuntu 22.04. Windows tool versions have been updated inline with Ubuntu tool updates.

### CODE REVIEW

File: ubuntu.22.04/Dockerfile [hunk 1]
```diff
@@ -70,7 +73,13 @@ RUN DOTNET_CLI_TELEMETRY_OPTOUT=1 && \
 RUN apt-get install -y openjdk-11-jdk-headless=${Java_Jdk_Version}
 
 # Install common Java tools
-RUN apt-get install -y maven gradle
+RUN apt-get update && \
+    apt-get install -y maven gradle
+
+# Get Libssl 1.1, required for Azure CLI as of Ubuntu 22.04. This can be removed when a proper Azure CLI ubuntu 22.04 version is avaliable
```
Comment (IsaacCalligeros95) [inline ubuntu.22.04/Dockerfile:79]: The Azure CLI needs a proper Ubuntu 22.04 release available. Using the old tool requires LibSSL 1.1, this is intended to be removed when a valid version is released.

File: ubuntu.22.04/Dockerfile [hunk 2]
```diff
@@ -58,7 +58,10 @@ RUN wget --quiet -O - https://raw.githubusercontent.com/kubernetes/helm/master/s
 
 # Get .NET SDK 6.0
 # https://docs.microsoft.com/en-us/dotnet/core/install/linux-package-manager-ubuntu-1804
+# https://learn.microsoft.com/en-us/dotnet/core/install/linux-package-mixup
 RUN DOTNET_CLI_TELEMETRY_OPTOUT=1 && \
+    touch /etc/apt/preferences && \
```
Comment (IsaacCalligeros95) [inline ubuntu.22.04/Dockerfile:63]: We need to ignore DotNet related packages from `packages.microsoft.com`, which now double up with the actual Ubuntu apt releases.

**Comment by IsaacCalligeros95** [comment]
I've kept the 18.04 Dockerfile but updated the cake and .sh build script to specifically look at 22.04. The intention is to keep this publicly available for anyone basing images off of this older release. Moving forward only 22.04 will be built and published for the ubuntu streams.

File: ubuntu.22.04/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+> Please note that we update this document periodically to match the latest version on DockerHub which is publicly available.
+> This does not necessarily match the content of Dockerfiles in this repository, as they may contain changes that are not released yet.
+
+## Image Name
+`octopusdeploy/worker-tools`
+
+## Tags
+
+- `3.3.2-ubuntu.22.04`
```
Comment (mik-ky) [inline ubuntu.22.04/README.md:11]: We won't have these tags built for the new ubuntu 22.04 right?

I would've thought only `4.0.0-ubuntu.22.04` (or w/e the next tag) would be available.

**Review by mik-ky — APPROVED** [review]
Looks good. Just if you could update the docs mentioned in the comment, it'd be great.

**Pull request merged**